### PR TITLE
Please check my comments on NER

### PR DESCRIPTION
**Pull request by dadelani**

### CODE REVIEW

File: named-entity-recognition/README.md [hunk 1]
```diff
@@ -13,16 +13,34 @@ There are some sources for labeled NER data that can be used to train systems.
 
 You can also create data yourself if it doesn't exist.
 
-TODO: A good procedure to do this should be filled in.
+* Start by collecting unlabelled texts from popular domains like News articles or Wikipedia. If unlabelled texts do not exist, you can translate some sentences.
```
Comment (neubig) [inline named-entity-recognition/README.md:16]: News
-> news

you can translate some sentences
-> you can translate some sentences, but you should try to get target-language text if at all possible because translated text can be unnatural

File: named-entity-recognition/README.md [hunk 2]
```diff
@@ -13,16 +13,34 @@ There are some sources for labeled NER data that can be used to train systems.
 
 You can also create data yourself if it doesn't exist.
 
-TODO: A good procedure to do this should be filled in.
+* Start by collecting unlabelled texts from popular domains like News articles or Wikipedia. If unlabelled texts do not exist, you can translate some sentences. 
+* Start with few sets of entities especially the popular ones like Personal names (PER), organization (ORG), location (LOC) and/or geopolitical entities (GPE). With few sets of entities, one can easily ensure high quality annotation since fine-grained entities and numerous entities can affect the quality of the overall process especially for non-expert annotators.
```
Comment (neubig) [inline named-entity-recognition/README.md:17]: Personal names
-> personal names

few sets of entities
-> a smaller set of entities

one can easily ensure high quality annotation
-> it is easier to maintain the quality of annotation

File: named-entity-recognition/README.md [hunk 3]
```diff
@@ -13,16 +13,34 @@ There are some sources for labeled NER data that can be used to train systems.
...
+* Familiarize yourself with annotation guides that have been used to develop popular NER datasets like the Message Understanding Conference [(MUC-6) guideline](https://cs.nyu.edu/faculty/grishman/NEtask20.book_1.html) and [LDC guideline](https://www.ldc.upenn.edu/sites/www.ldc.upenn.edu/files/english-entities-guidelines-v6.6.pdf)
+* Look for an easy to use annotation tool like [ELISA](https://aclanthology.org/P18-4001/), [brat](https://brat.nlplab.org/index.html), [Prodigy](https://prodi.gy/), [LightTag](https://www.lighttag.io/), [ioAnnotator](https://ioannotator.com/),  etc. Some provide the tool free for academic use.
+* Create your own annotation guide with examples in the language or domain of interest. Preferrably, with examples from the annotation tool.
+* (Optionally) Create a pre-recorded video on how to use the tool especially if you are working with non-expert annotators. 
+* Recruit odd number of annotators if possible, so that their inter-agreement score can be compared and you can make use of majority vote. Set your required inter-agreement score to be high (e.g at least 0.95) especially for the trial examples.
```
Comment (neubig) [inline named-entity-recognition/README.md:22]: odd number
-> an odd number

majority vote
-> majority voting

"Set your required inter-agreement score to be high" is a bit confusing. It's not clear what the "required inter-agreement score" is (and it's actually "inter-annotator agreement"). Maybe move this mention of inter-annotator agreement to the next bullet.

File: named-entity-recognition/README.md [hunk 4]
```diff
@@ -13,16 +13,34 @@ There are some sources for labeled NER data that can be used to train systems.
...
+* When annotation is done, prepare your final dataset in CoNLL format i.e <word>[space]<tag>, and sentences are demarcated with an additional empty line. An example is [here](https://github.com/masakhane-io/masakhane-ner/blob/main/data/pcm/dev.txt). The recommended tagging scheme is BIO or IOB2 where "B-" signifies the beginning of an entity, "I-" signifies the inside of an entity and "O"- outside of an entity or no entity. Other tagging schemes are IOBES and IOB1.  
 
 ## Training a Named Entity Recognition System
+  
+There are several NER models that can be trained, the best approach is fine-tuning pre-trained language models (PLMs) like BERT. 
 
-TODO: This should be filled in.
-
+You can use the official token classification code from [HuggingFace Transformers](https://github.com/huggingface/transformers/tree/master/examples/pytorch/token-classification). Fine-tuned LM produces the state-of-the-art for this task. For other languages apart from English, you need to use multilingual variants of the pre-trained LM e.g mBERT, XLM-RoBERTa, InfoXLM, RemBERT, etc.
```
Comment (neubig) [inline named-entity-recognition/README.md:30]: "Fine-tuned LM produces the state-of-the-art for this task." may be hard to understand if people don't understand "Fine-tuned LM".

File: named-entity-recognition/README.md [hunk 5]
```diff
@@ -13,16 +13,34 @@ There are some sources for labeled NER data that can be used to train systems.
...
 
 ## Training a Named Entity Recognition System
+  
+There are several NER models that can be trained, the best approach is fine-tuning pre-trained language models (PLMs) like BERT. 
 
-TODO: This should be filled in.
-
+You can use the official token classification code from [HuggingFace Transformers](https://github.com/huggingface/transformers/tree/master/examples/pytorch/token-classification). Fine-tuned LM produces the state-of-the-art for this task. For other languages apart from English, you need to use multilingual variants of the pre-trained LM e.g mBERT, XLM-RoBERTa, InfoXLM, RemBERT, etc. 
+  
+Please, confirm if your script is covered in the vocabulary of the PLM. If not, one idea is to perform [vocabulary augmentation](https://aclanthology.org/2020.findings-emnlp.118/) with the new script tokens and further pre-train the PLM on unlabelled texts in the new language.
```
Comment (neubig) [inline named-entity-recognition/README.md:32]: Please, confirm if your script
-> Confirm if the script the language is written in is covered

A little bit more info about concretely how to do this would be useful.

**Review by neubig — COMMENTED** [review]
Looks really great! I left some comments below, please take a look. This can be merged after you take a look at those.

File: named-entity-recognition/README.md (hunk 3 above)
Comment (dadelani) [reply named-entity-recognition/README.md:22]: changed inter-agreement score to inter-annotator agreement

the inter-annotator agreement metric is 0.95 Fleiss kappa score. I have updated it

File: named-entity-recognition/README.md (hunk 4 above)
Comment (dadelani) [reply named-entity-recognition/README.md:30]: Added brief description of fine-tuning: 

  
There are several NER models that can be trained, the best approach is fine-tuning pre-trained language models (PLMs) like BERT. Fine-tuning involves adding a linear classifier (that is randomly initialized) on top of PLM before training the joint model end-to-end (i.e jointly training PLM+Linear Classifier as a single model).
